prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -41,40 +41,94 @@ class FHIRRouter(
+        "%patient.birthDate.exists()",
+        "%specimen.type.exists()",
+        "(%patient.address.line.exists() or " +
+            "%patient.address.postalCode.exists() or " +
+            "%patient.telecom.exists())",
         "(" +
-            "(Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or " +
-            "Bundle.entry.resource.ofType(Specimen).collection.collected.exists()" +
+            "(%specimen.collection.collectedPeriod.exists() or " +
+            "%specimen.collection.collected.exists()" +
             ") or " +
-            "Bundle.entry.resource.ofType(ServiceRequest).occurrence.exists() or " +
-            "Bundle.entry.resource.ofType(Observation).effective.exists())"
+            "%serviceRequest.occurrence.exists() or " +
+            "%observation.effective.exists())"
     )
 
     /**
      * Default Rule:
      *  Must have a processing mode id of 'P'
      */
     val processingModeFilterDefault: ReportStreamFilter = listOf(
-        "Bundle.entry.resource.ofType(MessageHeader).meta" +
-            ".extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-processing-id')" +
-            ".value.coding.code = 'P'"
+        "%processingId = 'P'"
     )
 
+    /**
+     * Lookup table `fhirpath_filter_shorthand` containing all of the shorthand fhirpath replacements for filtering.
+     */
+    private val shorthandLookupTable by lazy { loadFhirPathShorthandLookupTable() }
+
+    /**
+     * Constants to make writing filter conditions shorter / more accessible. This will replace the shorthand
+     * used in configuration filter expressions with the specified Fhir Path expression before the expression
+     * is evaluated against the bundle. This allows for returning of collections, as well as handling 'exists()'
+     * where needed. Format is %myVariable within the filters, and each variable used must be defined
+     * as part of the shorthand collection of an exception will be raised. Make sure to verify if you need to use
+     * 'exists' on your check - an 'unable to parse' will return 'false' but will not raise any other error due to the
+     * underlying fhir library.
+     *
+     * The values used are located in the fhirpath_filter_shorthand lookup table.
+     */
+    private val regexVariable = """%[`']?[A-Za-z][\w\-'`_]*""".toRegex()
+    internal fun replaceShorthand(input: String): String {
+        var output = input
+        regexVariable.findAll(input)
+            .map { it.value }
+            .sortedByDescending { it.length }
+            .forEach {
+                val replacement = shorthandLookupTable[it.trimStart('%')]
+                if (!replacement.isNullOrEmpty()) {
+                    output = output.replace(it, replacement)
+                } else {
+                    throw NotImplementedError(
+                        "Could not locate a unique entry for '$it' in the shorthand values lookup table."
```

Comment:
This may bring a problem.  If I were to specify valid built-in FHIR Path variables like %resource or %rootresource or %ucum then will this error out?

I tested: `engine.replaceShorthand("%rootResource.exists() or %patientState or %patient")` and it gave an error. Will suggest to leave any variables you cannot substitute and the FHIR Path engine will do the rest.